File: gitdot-web/packages/api/src/endpoint/organization/list-organization-repositories.ts
import { z } from "zod";
import { page, RepositoryResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListOrganizationRepositoriesRequest = z.object({
  cursor: z.string().optional(),
  limit: z.number().int().positive().optional(),
});
export type ListOrganizationRepositoriesRequest = z.infer<
  typeof ListOrganizationRepositoriesRequest
>;

export const ListOrganizationRepositoriesResponse = page(RepositoryResource);
export type ListOrganizationRepositoriesResponse = z.infer<
  typeof ListOrganizationRepositoriesResponse
>;

export const ListOrganizationRepositories = {
  path: "/organization/{org_name}/repositories",
  method: "GET",
  request: ListOrganizationRepositoriesRequest,
  response: ListOrganizationRepositoriesResponse,
} as const satisfies Endpoint;
export type ListOrganizationRepositories = typeof ListOrganizationRepositories;

File: gitdot-api/src/endpoint/organization/list_organization_repositories.rs
use serde::{Deserialize, Serialize};

use crate::{
    endpoint::Endpoint,
    resource::{RepositoryResource, common::Page},
};

pub struct ListOrganizationRepositories;

impl Endpoint for ListOrganizationRepositories {
    const PATH: &'static str = "/organization/{org_name}/repositories";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListOrganizationRepositoriesRequest;
    type Response = ListOrganizationRepositoriesResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListOrganizationRepositoriesRequest {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub cursor: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub limit: Option<u32>,
}

pub type ListOrganizationRepositoriesResponse = Page<RepositoryResource>;
